components/raftstore/src/store/fsm/peer.rs

```diff
@@ -3686,6 +3689,24 @@ where
         }
 
         let mut meta = self.ctx.store_meta.lock().unwrap();
+        let is_region_initialized_in_meta = meta
+            .regions
+            .get(&region_id)
+            .map_or(false, |region| is_region_initialized(region));
+        if !is_initialized && is_region_initialized_in_meta {
+            let region_in_meta = meta.regions.get(&region_id).unwrap();
+            error!(
+                "peer is destroyed inconsistently";
+                "region_id" => region_id,
+                "peer_id" => self.fsm.peer_id(),
+                "merged_by_target" => merged_by_target,
+                "is_initialized" => is_initialized,
+                "is_region_initialized_in_meta" => is_region_initialized_in_meta,
+                "start_key" => log_wrappers::Value::key(region_in_meta.get_start_key()),
+                "end_key" => log_wrappers::Value::key(region_in_meta.get_end_key()),
+                "peers" => ?region_in_meta.get_peers(),
```

Comment:
should we print the peers in BOTH region_in_meta and self.region()? As this is the reason why is_region_initialized is different with  self.fsm.peer.is_initialized();